### PR TITLE
CAL-1361: Nothing appears on Calendar when creating a task with views Board

### PR DESCRIPTION
**Pull request by AzmiTouil**
when calendar integration is enabled in the project, the task in the view Board should also integrate by default.

### CODE REVIEW

File: task-management/src/main/java/org/exoplatform/task/management/controller/TaskController.java [hunk 1]
```diff
@@ -932,17 +932,15 @@ public Response createTaskInListView(String space_group_id, String taskTitle, Lo
       if (status != null) {
         task.setStatus(status);
       }
-    } else if (projectId != null) {
+    } if(projectId > 0) {
```
Comment (boubaker) [inline task-management/src/main/java/org/exoplatform/task/management/controller/TaskController.java]: Return to new line here, it's not related to an 'else'
+
Why deleting this condition ?

projectId is of type Long, thus, we should test on null and "> 0"

File: task-management/src/main/java/org/exoplatform/task/management/controller/TaskController.java [hunk 2]
```diff
@@ -932,17 +932,15 @@ public Response createTaskInListView(String space_group_id, String taskTitle, Lo
...
       Project project = projectService.getProject(projectId);
-      if (!project.canView(identity)) {
```
Comment (boubaker) [inline task-management/src/main/java/org/exoplatform/task/management/controller/TaskController.java]: Why deleting this check ?

File: task-management/src/main/java/org/exoplatform/task/management/controller/TaskController.java [hunk 3]
```diff
@@ -932,17 +932,15 @@ public Response createTaskInListView(String space_group_id, String taskTitle, Lo
...
-        throw new UnAuthorizedOperationException(projectId, Project.class, getNoPermissionMsg());
-      }
       Status status = statusService.getDefaultStatus(projectId);
-      if (status != null) {
-        task.setStatus(status);
```
Comment (boubaker) [inline task-management/src/main/java/org/exoplatform/task/management/controller/TaskController.java]: Why changing this assignment ?

File: task-management/src/main/java/org/exoplatform/task/management/controller/TaskController.java [hunk 4]
```diff
@@ -932,17 +932,15 @@ public Response createTaskInListView(String space_group_id, String taskTitle, Lo
...
+      if (status == null) {
+        throw new EntityNotFoundException(projectId, Project.class);
       }
+      task.setStatus(status);
+      task.setCalendarIntegrated(project.isCalendarIntegrated());
```
Comment (boubaker) [inline task-management/src/main/java/org/exoplatform/task/management/controller/TaskController.java]: The only useful instruction to fix this is this one:
```java
      task.setCalendarIntegrated(project.isCalendarIntegrated());
```

File: task-management/src/main/java/org/exoplatform/task/management/controller/TaskController.java (hunk 1 above)
Comment (AzmiTouil) [reply task-management/src/main/java/org/exoplatform/task/management/controller/TaskController.java]: done, but we need to delete the 'else' to make it work properly.

File: task-management/src/main/java/org/exoplatform/task/management/controller/TaskController.java (hunk 4 above)
Comment (AzmiTouil) [reply task-management/src/main/java/org/exoplatform/task/management/controller/TaskController.java]: Okay

**Pull request merged**